build.gradle

```diff
@@ -9,79 +9,221 @@
+    description 'Provide access control related features for OpenSearch'
+    classname 'org.opensearch.security.OpenSearchSecurityPlugin'
+}
+
+// This requires an additional Jar not published as part of build-tools
+loggerUsageCheck.enabled = false
+
+// No need to validate pom, as we do not upload to maven/sonatype
+validateNebulaPom.enabled = false
+
+publishing {
+    publications {
+        pluginZip(MavenPublication) { publication ->
+            pom {
+                name = "opensearch-security"
+                description = "Provide access control related features for OpenSearch"
+                licenses {
+                    license {
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                    }
+                }
+                developers {
+                    developer {
+                        name = "OpenSearch"
+                        url = "https://github.com/opensearch-project/security"
+                    }
+                }
+            }
+        }
+    }
+}
 
 repositories {
-    mavenLocal()
-    maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
     mavenCentral()
+    mavenLocal()
     maven { url "https://plugins.gradle.org/m2/" }
+    maven { url "https://aws.oss.sonatype.org/content/repositories/snapshots" }
     maven { url "https://d1nvenhzbhpy0q.cloudfront.net/snapshots/lucene/" }
 }
 
-ext {
-    isSnapshot = "true" == System.getProperty("build.snapshot", "true")
-    opensearch_version = System.getProperty("opensearch.version", "2.1.0-SNAPSHOT")
-    buildVersionQualifier = System.getProperty("build.version_qualifier", "")
-    version_tokens = opensearch_version.tokenize('-')
-    opensearch_build = version_tokens[0] + '.0'
-    if (buildVersionQualifier) {
-        opensearch_build += "-${buildVersionQualifier}"
-        opensearch_build_nosnapshot = opensearch_build
+tasks.withType(Checkstyle) {
+    showViolations true
+    reports {
+        ignoreFailures = false
     }
-    if (isSnapshot) {
-        opensearch_build += "-SNAPSHOT"
+}
+
+tasks.withType(JavaCompile) {
+    configure(options) {
+        options.encoding = 'UTF-8'
+        options.compilerArgs << '-Xlint:removal' << '-Werror'
+    }
+}
+
+tasks.test.finalizedBy(jacocoTestReport)  // report is always generated after tests run
+tasks.jacocoTestReport.dependsOn(test) // tests are required to run before generating the report
+
+
+allprojects {
+    tasks.withType(Javadoc).all { enabled = false }
+}
+
+bundlePlugin {
+    from('plugin-security.policy')
+    from('config') {
+        into 'config'
+    }
+    from('tools') {
+        into 'tools'
     }
 }
 
 configurations.all {
     resolutionStrategy {
         force 'commons-codec:commons-codec:1.14'
-        force 'org.apache.santuario:xmlsec:2.2.3'
-        force 'org.cryptacular:cryptacular:1.2.4'
-        force 'net.minidev:json-smart:2.4.7'
```

Comment:
have these libraries been removed to align with core and/or is it because we no longer need them?

Reply:
okie...got it